#32 Формирование отчета в формате Generic Execution (Sonar)

arkuznetsov opened the pull request.

Commit:
Реализован тестовый вариант формирование отчета в формате Generic Execution (sonar)

arkuznetsov committed:
Рефакторинг доработки вывода результатов тестирования в формате generic execution

arkuznetsov committed:
В gitignore добавлен файл настроек отладчика VSC

arkuznetsov committed:
Добавлен расчет времени выполнения тестовых процедур

arkuznetsov committed:
Исправлены ошибки регистрации времени выполнения тестов

arkuznetsov committed:
Исправлен формат вывода времени выполнения тестов в файл отчета

artbear commented:
@arkuznetsov Очень интересно, спасибо!

Пока некогда смотреть :(
работа + подготовка к конференции.

Очень постараюсь посмотреть и выдать обратную связь

artbear commented on testrunner.os:
```diff
@@ -793,9 +906,8 @@
-				ИЛИ Найти(текстОшибки, "Object method not found (") > 0)
-			И (Найти(текстОшибки, ИмяМетода) > 0)  Тогда
+	Если Найти(текстОшибки, "Метод объекта не обнаружен ("+имяМетода+")") > 0 
+		ИЛИ Найти(текстОшибки, "Object method not found ("+имяМетода+")") > 0  Тогда
```
Откати, плиз, эти строки
```bsl
Если Найти(текстОшибки, "Метод объекта не обнаружен ("+имяМетода+")") > 0 
+		ИЛИ Найти(текстОшибки, "Object method not found ("+имяМетода+")") > 0  Тогда
```
т.к. этот старый вариант неверно работает в 20 релизе :(

arkuznetsov replied:
Сорри, не понимаю, куда откатить? Я подлил к себе из твоего актуального девелопа там все на месте.

artbear commented on testrunner.os:
```diff
@@ -925,6 +1053,8 @@
 Процедура Инициализация()
     Лог = Логирование.ПолучитьЛог(ИмяЛога());
 
+	ФорматЛогФайла = "JU";
```
+ Значение `"JU"` нигде в коде не используется :( 
  + Есть проверка на `GE`
  + Зачем `"JU"` вообще тогда нужно?
+ Предлагаю вместо магических строк (`GE` и `JU`) заюзать фиксированную структуру и обращаться к ее свойствам
  + Например, `ВидыФорматаЛогФайла.JUnit` или `ВидыФорматаЛогФайла.GenericSonar`

artbear replied:
Примеры есть в 1тестраннере.
Например, `ЗначенияСостоянияТестов`

arkuznetsov replied:
Добавил структуру с форматами.
В структуре оставил значение JUnit, а также назначение значения по умолчанию для дальнейшего развития.

artbear commented on testrunner.os:
```diff
@@ -567,6 +572,97 @@
+	ЗаписьXML = Новый ЗаписьXML;
+	ЗаписьXML.ОткрытьФайл(ПутьОтчетаВФорматеJUnitxml.ПолноеИмя);
+	ЗаписьXML.ЗаписатьБезОбработки(СтрокаХМЛ);
+	ЗаписьXML.Закрыть();
```
Код дублируется с аналогичным методом в формировании отчета Junit.
Копипаст = зло :)
```bsl
ЗаписьXML = Новый ЗаписьXML;
+	ЗаписьXML.ОткрытьФайл(ПутьОтчетаВФорматеJUnitxml.ПолноеИмя);
+	ЗаписьXML.ЗаписатьБезОбработки(СтрокаХМЛ);
+	ЗаписьXML.Закрыть();
```

artbear replied:
Предлагаю выделить отдельный метод для этих строк

arkuznetsov replied:
Поправил.

artbear commented on testrunner.os:
```diff
@@ -567,6 +572,97 @@
 
+Процедура ЗавершитьЗаписьВФайлОтчетаТестированияВФорматеGExecXML(ЗаписьXML, ДатаНачала)
+	
+	Утверждения.ПроверитьНеРавенство(НаборТестов.Количество(), 0);
```
Предлагаю расшифровать утверждение, чтобы было понятно, в каком месте кода утверждение упало.
Например, 
```bsl
Утверждения.ПроверитьНеРавенство(НаборТестов.Количество(), 0, "Количество тестов равно нуля, что неверно. Метод ЗавершитьЗаписьВФайлОтчетаТестированияВФорматеGExecXML ");
```

arkuznetsov replied:
Поправил.

artbear reviewed (changes requested):
Отличное дополнение.
Но есть замечания.
@arkuznetsov Исправишь?
затем я приму.

ЗЫ иначе irac на штатном 1testrunner выполнить нельзя :(

arkuznetsov committed:
Merge branch 'develop' of https://github.com/artbear/1testrunner into develop

arkuznetsov committed:
Поправлены замечания к PR #33;
Добавлена структура с перечислением доступных форматов файла отчета;
Запись файла отчета вынесена в отдельную процедуру;
Мелкие исправления оформления кода и коментариев

arkuznetsov committed:
#33 Исправлено значение формата лог файла по умолчанию

arkuznetsov commented:
@artbear А по что у тебя сборка на трависе поломана?

artbear commented on testrunner.os:
```diff
@@ -773,9 +898,8 @@
-				ИЛИ Найти(текстОшибки, "Object method not found (") > 0)
-			И (Найти(текстОшибки, ИмяМетода) > 0)  Тогда
+	Если Найти(текстОшибки, "Метод объекта не обнаружен (" + имяМетода + ")") > 0 
+		ИЛИ Найти(текстОшибки, "Object method not found (" + имяМетода + ")") > 0  Тогда
```
Как я уже писал, откати эти строки, оставь, как у меня сейчас написано.

artbear replied:
Иначе на новых релизах движка будут ошибки

arkuznetsov replied:
Так вот я  не пойму как у тебя сейчас :-(. Я начинающий волшебник :-).
У меня в строках 777-778 вообще что-то другое, в 901-902 вроде так и есть.
И у меня на 20 крутится.

arkuznetsov replied:
Так, вроде въехал. Не привык я еще различия нормально читать.

arkuznetsov committed:
#33 Поправлена проверка ошибки отсутствия метода объекта
